nums/api.py

```diff
@@ -22,9 +22,12 @@
 
 def read(filename: str) -> BlockArray:
     """
-    :param filename: The name of the file to read. This must be the name of an array
-    that was previously written using the nums.write command.
-    :return: An instance of BlockArray.
+    Args:
+        filename (str): The name of the file to read. This must be the name of an array
```

Comment:
What do you think about omitting the type here for args, and enforcing static typing instead? This would be consistent with what they have for examples here: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings